packages/block-directory/src/store/controls.js

```diff
@@ -1,153 +1,71 @@
 /**
- * External dependencies
- */
-import { forEach } from 'lodash';
-
-/**
- * WordPress dependencies
- */
-import { createRegistryControl } from '@wordpress/data';
-import wpApiFetch from '@wordpress/api-fetch';
-
-/**
- * Calls a selector using the current state.
- *
- * @param {string} storeName    Store name.
- * @param {string} selectorName Selector name.
- * @param {Array}  args         Selector arguments.
- *
- * @return {Object} Control descriptor.
- */
-export function select( storeName, selectorName, ...args ) {
-	return {
-		type: 'SELECT',
-		storeName,
-		selectorName,
-		args,
-	};
-}
-
-/**
- * Calls a dispatcher using the current state.
- *
- * @param {string} storeName      Store name.
- * @param {string} dispatcherName Dispatcher name.
- * @param {Array}  args           Selector arguments.
- *
- * @return {Object} Control descriptor.
- */
-export function dispatch( storeName, dispatcherName, ...args ) {
-	return {
-		type: 'DISPATCH',
-		storeName,
-		dispatcherName,
-		args,
-	};
-}
-
-/**
- * Trigger an API Fetch request.
+ * Loads a JavaScript file.
  *
- * @param {Object} request API Fetch Request Object.
+ * @param {string} asset The url for this file.
  *
- * @return {Object} Control descriptor.
+ * @return {Promise} Promise which will resolve when the asset is loaded.
  */
-export function apiFetch( request ) {
-	return {
-		type: 'API_FETCH',
-		request,
-	};
-}
-
-/**
- * Loads JavaScript
- *
- * @param {Array}    asset   The url for the JavaScript.
- * @param {Function} onLoad  Callback function on success.
- * @param {Function} onError Callback function on failure.
- */
-const loadScript = ( asset, onLoad, onError ) => {
+export const loadScript = ( asset ) => {
 	if ( ! asset ) {
-		return;
+		return Promise.reject( new Error( 'No script found.' ) );
 	}
-	const existing = document.querySelector( `script[src="${ asset.src }"]` );
-	if ( existing ) {
-		existing.parentNode.removeChild( existing );
-	}
-	const script = document.createElement( 'script' );
-	script.src = typeof asset === 'string' ? asset : asset.src;
-	script.onload = onLoad;
-	script.onerror = onError;
-	document.body.appendChild( script );
+	return new Promise( ( resolve, reject ) => {
+		const existing = document.querySelector( `script[src="${ asset }"]` );
+		if ( existing ) {
+			existing.parentNode.removeChild( existing );
+		}
```

Comment:
Why does this matter?

Reply:
TBH It was in the original code and I didn't remove it. I think it's used when retrying loading assets, so we don't end up with duplicate script tags in the document.

Reply:
right.... to prevent issues during the injection step on retry. ok.